### PR TITLE
update current community when viewing a user

### PR DESCRIPTION
**Pull request by robbiecarlton**

### CODE REVIEW

**Comment by robbiecarlton** [comment]
@levity - good to merge?

File: src/containers/person/PersonProfile.js [hunk 1]
```diff
@@ -47,9 +47,15 @@ const initialFetch = (id, query) => {
...
 const PersonProfile = compose(
-  prefetch(({ dispatch, params: { id }, query }) =>
+  prefetch(({ store, dispatch, params: { id }, query }) =>
   Promise.all([
-    dispatch(fetchPerson(id)),
+    dispatch(fetchPerson(id))
+    .then(({ payload: {shared_communities} }) => {
+      let { currentCommunityId } = store.getState()
+      if (includes(shared_communities, currentCommunityId)) return
+      if (isEmpty(shared_communities)) return dispatch(setCurrentCommunityId('all'))
```
Comment (levity) [inline src/containers/person/PersonProfile.js]: more concisely:

``` js
return dispatch(setCurrentCommunityId(shared_communities[0] || 'all'))
```

File: src/containers/person/PersonProfile.js [hunk 2]
```diff
@@ -47,9 +47,15 @@ const initialFetch = (id, query) => {
 }
 
 const PersonProfile = compose(
-  prefetch(({ dispatch, params: { id }, query }) =>
+  prefetch(({ store, dispatch, params: { id }, query }) =>
   Promise.all([
-    dispatch(fetchPerson(id)),
+    dispatch(fetchPerson(id))
+    .then(({ payload: {shared_communities} }) => {
```
Comment (levity) [inline src/containers/person/PersonProfile.js]: spaces around `shared_communities` please

Comment (levity) [inline src/containers/person/PersonProfile.js]: also, try loading a profile with an invalid user id -- you may need to short-circuit return here if action.error is set and only get shared_communities afterward

**Pull request merged**